### PR TITLE
WebTorrent: Add "Save All Files..." feature

### PR DESCRIPTION
**Pull request by feross**
Currently there is no ability to save all files at once. The user has to save each file separately. This PR fixes that.

Fixes: https://github.com/brave/brave-browser/issues/1230

## Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/brave-browser/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [ ] Added/updated tests for this change (for new code or code which already has tests).
- Verified that these changes build without errors on
  - [ ] Android
  - [ ] iOS
  - [ ] Linux
  - [ ] macOS
  - [ ] Windows
- Verified that these changes pass automated tests (unit, browser, security tests) on
  - [ ] iOS
  - [ ] Linux
  - [ ] macOS
  - [ ] Windows
- [x] Verified that all lint errors/warnings are resolved (`npm run lint`)
- [x] Ran `git rebase master` (if needed).
- [x] Ran `git rebase -i` to squash commits (if needed).
- [x] Tagged reviewers and labelled the pull request as needed.
- [ ] Request a security/privacy review as needed.
- [x] Add appropriate QA labels (QA/Yes or QA/No) to include the closed issue in milestone
- [ ] Public documentation has been updated as necessary. For instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protection-Mode
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-compatibility-issues-with-tracking-protection
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide

## Test Plan:

1. Visit https://webtorrent.io/free-torrents
1. Click on the WIRED CD torrent
1. Click start download
1. Try to click on the disabled "Save All Files..." link *before* the download has completed. Confirm that it doesn't work.
1. Once the download is complete, click on "Save All Files...". It should prompt for a location to save the .zip file to.
1. Unzip the zip file however you do that on your OS and confirm that the torrent's files are inside.

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [ ] Adequate test coverage exists to prevent regressions 
- [ ] Verify test plan is specified in PR before merging to source

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on.
- [ ] All relevant documentation has been updated.

### CODE REVIEW

File: components/brave_webtorrent/extension/background/webtorrent.ts [hunk 1]
```diff
@@ -79,3 +81,64 @@ export const delTorrent = (infoHash: string) => {
 
   maybeDestroyWebTorrent()
 }
+
+export const saveAllFiles = (infoHash: string) => {
+  console.log('Time to save the file with jszip!', infoHash)
```
Comment (yrliou) [inline components/brave_webtorrent/extension/background/webtorrent.ts]: unnecessary debug log?

File: package.json [hunk 2]
```diff
@@ -328,11 +328,13 @@
     "webpack-cli": "^3.0.8"
   },
   "dependencies": {
+    "@types/jszip": "^3.1.6",
     "@types/parse-torrent": "^5.8.3",
     "@types/webtorrent": "^0.98.5",
     "bignumber.js": "^7.2.1",
     "bluebird": "^3.5.1",
     "clipboard-copy": "^2.0.0",
+    "jszip": "^3.2.2",
```
Comment (yrliou) [inline package.json]: cc @diracdeltas for adding a new dep here.

File: components/brave_webtorrent/extension/manifest.json [hunk 3]
```diff
@@ -7,7 +7,7 @@
     "scripts": ["extension/out/brave_webtorrent_background.bundle.js"],
     "persistent": true
   },
-  "permissions": ["dns", "tabs", "windows", "<all_urls>"],
+  "permissions": ["downloads", "dns", "tabs", "windows", "<all_urls>"],
```
Comment (yrliou) [inline components/brave_webtorrent/extension/manifest.json]: cc @diracdeltas for granting a new permission to webtorrent extension.

Comment (diracdeltas) [reply components/brave_webtorrent/extension/manifest.json]: seems ok

File: package.json (hunk 2 above)
Comment (diracdeltas) [reply package.json]: ++

File: components/brave_webtorrent/extension/background/webtorrent.ts (hunk 1 above)
Comment (feross) [reply components/brave_webtorrent/extension/background/webtorrent.ts]: Removed

Comment (feross) [reply components/brave_webtorrent/extension/background/webtorrent.ts]: Removed

**Comment by feross** [comment]
Force pushed with the stray `console.log` removed.

**Comment by yrliou** [comment]
I'm going to merge it here without waiting for CI again since the latest force push only had a `console.log` removed and things are already green before this force push.

**Pull request merged**